### ISSUE TITLE
0 for Null Aggregates

### ISSUE THREAD

**Issue by sblatnick** [kairosdb/kairosdb#359]
It would be very beneficial for visualization software to create an api argument or setting allowing for aggregators to treat missing data points for a time period as 0 or null, instead of just skipping a point.

In other words, I would expect when aggregating by time, that every time point would have a value, even if zero.

Please see my description of the problem in https://github.com/grafana/kairosdb-datasource/issues/27

They suggested placing this request here.

**Comment by lcoulet** [comment]
Hi, this is the purpose of the gaps marking aggregator.
0 doesn't make sense since there's no way to know if that's a real 0 or a
data loss, but null does and is handled properly by grafana and kairosdb UI.

Loic

Le 30 janv. 2017 9:09 PM, "sblatnick" <notifications@github.com> a écrit :

It would be very beneficial for visualization software to create an api
argument or setting allowing for aggregators to treat missing data points
for a time period as 0 or null, instead of just skipping a point.

In other words, I would expect when aggregating by time, that every time
point would have a value, even if zero.

Please see my description of the problem in grafana/kairosdb-datasource#27
<https://github.com/grafana/kairosdb-datasource/issues/27>

They suggested placing this request here.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/kairosdb/kairosdb/issues/359>, or mute the thread

**Comment by brianhks** [comment]
Loic is correct, the gaps agg is what you are after.
